src/backend/utils.ts

```diff
@@ -889,13 +889,7 @@ export const spawnAsync = async (
   }
 
   return new Promise((resolve, reject) => {
-    child.on('error', (error) =>
-      reject({
-        code: 1,
-        stdout: stdout.join(''),
-        stderr: stderr.join('').concat(error.message)
-      })
-    )
+    child.on('error', reject)
```

Comment:
I'm curious about this, I understand it's so the `reject` receives an Errors object, but it seems to have a different signature than the one we were sending before, is there a place that catches this rejections that should be updated somehow?

I looked for places calling `spawnAsync` and I feel like we've had this wrong all this time, from how it's used it looks like it was intended to be

```ts
child.on('error', (error) =>
  resolve({
    code: 1,
    stdout: stdout.join(''),
    stderr: stderr.join('').concat(error.message)
  })
)
```

(using `resolve` instead of `reject`)

Reply:
I did come to the same conclusion as you did (`resolve` being the intended function call), but I didn't want to break any existing behavior that relied on the promise being rejected (if such code exists anywhere). Changing the type of the rejected value is relatively safe to do (since the type of it is going to be `unknown` to consumer functions), changing the behavior from "rejects on error" to "resolves on error" might not be

Reply:
maybe we can add a `TODO` and keep the old code commented out? to not forget about this, I don't know haha

I do think it's a bug we have, but I guess it's failing with an exception when the spawn fails, instead of handling the error, though it's an error state in both cases

maybe it's just too rare for that to actually fail and it probably means other things failed before anyway and that's why we never noticed it